syntax = "proto3";

import "google/protobuf/any.proto";

message BoundingBox {
    int32 x_min = 1;
    int32 y_min = 2;
    int32 x_max = 3;
    int32 y_max = 4;
}

message ImageRequest {
    message Image {
        string image_path = 1;
        repeated BoundingBox bboxes = 3;
        google.protobuf.Any points = 4;
        google.protobuf.Any labels = 5;
    }
    repeated Image images = 1;
}

message DictionaryEntry {
    string image_path = 1;
    bool processed = 2;
    string png_result_path = 3;
    string tif_result_path = 4;
}
message ImageResponse {
    repeated DictionaryEntry entries = 1;
}

service ImageProcessor {
    rpc ProcessImage(ImageRequest) returns (ImageResponse);
}
